### PR TITLE
v1.2.1 Gopay tokenization

### PR DESCRIPTION
**Pull request by Xaxxis**
Feature:
• add Gopay tokenization API

Fix Bug:
• fix some typo
• fix nill config option in subscription method/function

### CODE REVIEW

File: coreapi/request.go [hunk 1]
```diff
@@ -244,3 +248,14 @@ type Schedule struct {
 	// start_time will be current time and first payment will happen on one interval after current time.
 	StartTime string `json:"start_time,omitempty"`
 }
+
+type PaymentAccount struct {
```
Comment (rizdaprasetya) [inline coreapi/request.go]: Should be `PaymentAccountReq` to maintain naming consistency.

File: coreapi/request.go [hunk 2]
```diff
@@ -244,3 +248,14 @@ type Schedule struct {
...
+	PaymentType  CoreapiPaymentType `json:"payment_type"`  // Payment channel where the account register to
+	GopayPartner *GopayPartner      `json:"gopay_partner"` // GoPay linking specific parameters
+}
+
+type GopayPartner struct {
```
Comment (rizdaprasetya) [inline coreapi/request.go]: Should be `GopayPartnerDetails`

File: coreapi/request.go [hunk 3]
```diff
@@ -244,3 +248,14 @@ type Schedule struct {
 	// start_time will be current time and first payment will happen on one interval after current time.
 	StartTime string `json:"start_time,omitempty"`
 }
+
```
Comment (rizdaprasetya) [inline coreapi/request.go:251]: Also should rename type `Schedule`  to `ScheduleDetails` to maintain naming consistency.

File: coreapi/response.go [hunk 4]
```diff
@@ -230,3 +231,33 @@ type ScheduleResponse struct {
...
+	AccountStatus          string   `json:"account_status"`
+	ChannelResponseCode    string   `json:"channel_response_code"`
+	ChannelResponseMessage string   `json:"channel_response_message"`
+	Action                 Action   `json:"action"`
+	Metadata               Metadata `json:"metadata"`
+	StatusMessage          string   `json:"status_message"`
+	ID                     string   `json:"id"`
+}
+
+type Metadata struct {
```
Comment (rizdaprasetya) [inline coreapi/response.go]: Better name the type `Metadata` as `PaymentAccountMetadataDetails`.

So the struct field become:
```
Metadata               PaymentAccountMetadataDetails `json:"metadata"`
```

File: coreapi/response.go [hunk 5]
```diff
@@ -230,3 +231,33 @@ type ScheduleResponse struct {
...
+	PaymentOptions []PaymentOptions `json:"payment_options"`
+}
+
+type PaymentOptions struct {
```
Comment (rizdaprasetya) [inline coreapi/response.go]: should be `PaymentOptionsDetails`

File: coreapi/response.go [hunk 6]
```diff
@@ -230,3 +231,33 @@ type ScheduleResponse struct {
...
+	Name     string      `json:"name"`
+	Active   bool        `json:"active"`
+	Metadata interface{} `json:"metadata"`
+	Balance  Balance     `json:"balance"`
+	Token    string      `json:"token"`
+}
+
+type Balance struct {
```
Comment (rizdaprasetya) [inline coreapi/response.go]: Should be `BalanceDetails`

**Review by rizdaprasetya — CHANGES_REQUESTED** [review]
👍 Overall looks good. Just some naming consistency need to be fixed. Thanks!

File: coreapi/request.go (hunk 1 above)
Comment (Xaxxis) [reply coreapi/request.go]: done

File: coreapi/request.go (hunk 2 above)
Comment (Xaxxis) [reply coreapi/request.go]: done

File: coreapi/request.go (hunk 3 above)
Comment (Xaxxis) [reply coreapi/request.go:251]: done

File: coreapi/response.go (hunk 4 above)
Comment (Xaxxis) [reply coreapi/response.go]: done

File: coreapi/response.go (hunk 5 above)
Comment (Xaxxis) [reply coreapi/response.go]: done

File: coreapi/response.go (hunk 6 above)
Comment (Xaxxis) [reply coreapi/response.go]: done

**Review by rizdaprasetya — APPROVED** [review]
Nice, looks good now. Good to merge & release. Thanks @Xaxxis ! 🎉

**Pull request merged**